### DISCUSSION TITLE
deck.gl very power hungry when idle 🤤

### DISCUSSION THREAD

**Discussion by MrMorice** [visgl/deck.gl#9041] [Q&A]
I love deck.gl but is there any way to reduce power draw when idle (currently 50% CPU on a MacBook Pro / Safari for simple examples like https://deck.gl/examples/icon-layer)? Ideal would be to freeze the current context and only change when triggered. This would be in line with how maplibre performs.

For my specific app, the idle power draw for everything except deck.gl is at roughly 2% CPU and with deck.gl around 40-60%. This is a big issue with mobile users.

Attached a screenshot of the Safari Timeline Power Usage for completely idle state.

<img width="740" alt="Screenshot 2024-07-23 at 19 04 33" src="https://github.com/user-attachments/assets/36416a12-46a9-4543-bdd5-2200a385933e">

**Comment by ibgreen** [comment]
That looks like either a bug in deck, or some application specific issue. deck should not rerender unless something has changed. 

- Are you able to pinpoint in the flamegraph what is happening?
- Are you able to reproduce this with deck.gl website examples?

**Reply by MrMorice** [reply]
Hi @ibgreen, I was not able to pinpoint the problem but was able to reproduce this on any of the website examples. After complete render the usage stays at roughly 50%. E.g. https://deck.gl/examples/icon-layer or https://deck.gl/examples/line-layer.

Besides this, deck.gl is an amazing library and really works great for my users. Thank you for the great job!

**Reply by MrMorice** [reply]
FYI: Now created a bug report for tracking.

**Reply by bobdepreziz** [reply]
Hi @MrMorice can you link me the bug report, or is it not a bug created on github? I noticed to that our deck.gl map continuously uses CPU while idling and nothing is happening. Thanks.

**Reply by MrMorice** [reply]
Hey @bobdepreziz, thanks for checking in. The related bug is https://github.com/visgl/deck.gl/issues/9084
